pkg/auth/azure.go

```diff
@@ -161,17 +161,26 @@ func HttpAuthMiddleWare(resp http.ResponseWriter, req *http.Request, jwks *keyfu
 			return nil
 		}
 	}
-	// Skip azure authentication with ID for `/release`, `/releasetrain` and `/locks`  endpoints. The requests will be validated with pgp signature
+
+	// Skip azure authentication with ID for `/` (POST: createEnv), `/release`, `/releasetrain` and `/locks`  endpoints. The requests will be validated with pgp signature
 	// usage in requests from outside the cluster (e.g. by GitHub Actions and the publish.sh script).
-	releaseTrainRx := regexp.MustCompile("^/environments/[^/]+/releasetrain$")
-	if releaseTrainRx.MatchString(req.URL.Path) {
-		return nil
-	}
-	environmentLocksRx := regexp.MustCompile("^/environments/[^/]+/locks/[^/]+$")
-	if environmentLocksRx.MatchString(req.URL.Path) {
-		return nil
+	group, tail := xpath.Shift(req.URL.Path)
```

Comment:
As I understand, this is shifting something in the request, but I don't understand what. Can you explain a bit @medo-freiheit?

Reply:
From what I understand the shift simply splits the leftmost part of the URL
So if you have `/environments/prod/blabla`, the shift will `environments` part in the `group` variable and the `/prod/blabla` into the `tail` variable

Reply:
Alright, I am satisfied with the answer! Looks good to me